### PR TITLE
Fixed preplanning controller crash

### PR DESCRIPTION
**Pull request by steam-test1**
# Description

When you use a controller and click delete plan when there is only one item saved it crashes

Fixes issues #746

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tested with the latest commit

# Checklist:

- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by Kamikaze94** [comment]
Hi, 
Thanks for this bugfix. 

Please avoid hard-overwriting functions when possible. Store the original function in a local variable instead and call that one inside the function, surrounded by your new checks or manipulation of in and output variables. It helps compatability with other mods as well as potential future updates by Overkill. Another good practice is adding the three dots to the function declaration and call.

In this case, you would do:
```lua
local set_map_position_by_item_original = PrePlanningMapGui.set_map_position_by_item
function PreplanningMapGui:set_map_position_by_item(item, ...)
    if item then
        return set_map_position_by_item_original(item, ...)
    end
end
```

I'd be happy if you can change this, before merging.

**Comment by steam-test1** [comment]
Done
I was going to do that, but forgot

File: lua/PrePlanManager.lua [hunk 1]
```diff
@@ -670,4 +670,10 @@ elseif requiredScript == "lib/managers/preplanningmanager" then
 			QuickMenu:new(managers.localization:text("wolfhud_preplanning_manager_title"), message, { text = managers.localization:text("dialog_ok"), is_cancel_button = true }, true)
 		end
 	end
+elseif requiredScript == "lib/managers/menu/preplanningmapgui" then
+	local set_map_position_to_item_original = PrePlanningMapGui.set_map_position_to_item
+	function PrePlanningMapGui:set_map_position_to_item(...)
+	if item then
+		return set_map_position_to_item_original(self, ...)
+	end
```
Comment (Kamikaze94) [inline lua/PrePlanManager.lua]:
```lua
function PrePlanningMapGui:set_map_position_to_item(item, ...)
	if item then
		return set_map_position_to_item_original(self, item, ...)
	end
end
```

Comment (Kamikaze94) [reply lua/PrePlanManager.lua]: Variable item needs to be defined and passed on.
'end'-statement for if-clause missing.

**Comment by steam-test1** [comment]
It should to fixed now

**Pull request merged**